### ISSUE TITLE
 align "┌ ─ ┐ │ └ ┘"

### ISSUE THREAD

**Issue by bo-ku-ra** [terrastruct/d2#860]
https://play.d2lang.com/?l=&script=pJG7SsVAEIb7eYq_jEVkZ-87Qp4kjZiAFtlAiIjIvLusgnjK7Gm-KYbvZy5_xxYs-0clxjjBNrgG3xAotpIackPBOLEhFvBch7qfeHk-1gd8_Qz0ePMqwXm8r0_4bf3bTMBKVmDvibBKTuDmOrxu22XbKXmB77S9UhCETjsoRUGc67C8HefnZT8qJUHq9pNSFuRuPysVQen2ixIbAZvuBDZK3wEAAP__&

**Comment by gavin-ts** [comment]
Is the issue that the double border stroked dash lines don't line up (due to the outer path being longer than the inner one?

![Screen Shot 2023-02-20 at 7 05 39 PM](https://user-images.githubusercontent.com/85081687/220237526-0bbcd0e3-1bdd-4276-806a-b8d630b88d03.png)

Is the idea to have it lined up more like this (but with equal lengths)?
![Screen Shot 2023-02-20 at 7 09 06 PM](https://user-images.githubusercontent.com/85081687/220237798-bcfc5071-9f94-4d6d-a3d5-b50c83167e1c.png)

**Comment by bo-ku-ra** [comment]
i think

best: 
all corners must be "┌  ┐ └ ┘".

better:
the inner line is shorter than the outer line.
or
your solution.

**Comment by alixander** [comment]
i think we have to use border-image for this lol. https://developer.mozilla.org/en-US/docs/Web/CSS/border-image

i'll tag as low-priority because that's just how CSS is. but if someone wants to try it, be my guest

**Comment by Nepo26** [comment]
Hi, I want to tackle this issue so I start to understand the codebase. Is it ok?

**Comment by alixander** [comment]
@Nepo26 no, when it's parent-child, it's far apart enough that no one will notice. `double-border is when it's noticeable: 

https://play.d2lang.com/?script=qqhMS7NSqOZSUAAbqIcSblYKJUWlqXA5JJdYKRhx1XIBAgAA__8%3D&

**Comment by bo-ku-ra** [comment]
https://play.d2lang.com/?script=qoCCSghIgwErhWouBQWwJXooYWmlUFJUmgqXQ3KdlYKhgYKygoGenqEBVy0XIAAA__8%3D&

**Comment by Nepo26** [comment]
So hey. I looked more into this issue. And this is what I found and how I am trying to make it work.

- I looked more into [border-image](https://developer.mozilla.org/en-US/docs/Web/CSS/border-image) and it don't seem like the correct solution as it uses an image.
- Tried to manually find the perfect ratio of **_stroke-dasharray_** and **_stroke-dashoffset_**, didn't get far. I was able to get a corner created if the offset was `width / height` but it only worked if the **_stroke-width_** was above 15 and it looked ugly.
- Found out that they were introducing a new property called [**_stroke-dashcorner_**](https://www.w3.org/TR/svg-strokes/#StrokeDashcornerProperty) that would be perfect. Seems like it's yet premature to be used.
- Then I saw a [twitter thread](https://twitter.com/steveruizok/status/1527971784603095041) of someone in [tldraw](https://www.tldraw.com/) who just fixed a similar problem to ours. Looking at their source code I saw that they instead of using a **_rect_** property they use grouped lines.

## In the end what am I doing?
I'm trying to replicate the [**DashedOutlineBox**](https://github.com/tldraw/tldraw/blob/main/packages/editor/src/lib/editor/shapes/shared/DashedOutlineBox.tsx) and the [**getPerfectDashProps](**https://github.com/tldraw/tldraw/blob/main/packages/editor/src/lib/editor/shapes/shared/getPerfectDashProps.ts) to allow us the to create better boxes.

**Comment by alixander** [comment]
@Nepo26 nice find! 

indeed, tldraw's dash treatment is 👨‍🍳 👌  . steve is a legend.

<img width="252" alt="Screen Shot 2023-06-28 at 1 51 46 PM" src="https://github.com/terrastruct/d2/assets/3120367/7c404738-bf43-42b1-9045-7580dde4d3b9">
